capstone/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,44 @@
+package com.google.sps.servlets;
+import java.io.IOException;
+import java.io.BufferedReader;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.gson.Gson;
+import com.google.sps.data.Form;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        BufferedReader reader = request.getReader();
+        StringBuilder sb = new StringBuilder();
+        try {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                sb.append(line).append('\n');
+            }
+        } finally {
+            reader.close();
+        }
+        Gson gson = new Gson();
+        Form form = gson.fromJson(sb.toString(), Form.class);
+        
+        Entity formEntity = form.toDatastoreEntity();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(formEntity);
+    }
```

Comment:
Have yet to add a response. Will probably be a success/fail and after, form-loader.js would fetch and call the doGet method.

Reply:
Does it still respond 200 OK? Even if there is no content in the response yet, it should still respond something

Reply:
Alright, will make it so that it responds with 200 OK if it works.